compiler/src/dotty/tools/dotc/transform/SymUtils.scala

```diff
@@ -259,6 +259,22 @@ object SymUtils:
         && self.owner.linkedClass.is(Case)
         && self.owner.linkedClass.isDeclaredInfix
 
+    /** Is symbol declared or inherits @experimental? */
+    def isExperimental(using Context): Boolean =
+      def isExperimentalOverloaded: Boolean =
+        self.allOverriddenSymbols.nonEmpty
+        && self.allOverriddenSymbols.forall(_.hasAnnotation(defn.ExperimentalAnnot)) // TODO infer or enforce annotation?
+
+      self.eq(defn.ExperimentalAnnot)
+      || self.hasAnnotation(defn.ExperimentalAnnot)
+      || isExperimentalByNesting
+      || isExperimentalOverloaded
+
+    /** Is symbol is nested in a @experimental definition? */
+    def isExperimentalByNesting(using Context): Boolean =
+      !self.isRefinementClass && // FIXME: avoid cyclic reference errors on poly functions
```

Comment:
@smarter have you encountered something similar before?

Reply:
I don't think so